paymentsheet/src/main/java/com/stripe/android/paymentsheet/flowcontroller/FlowControllerConfigurationHandler.kt

```diff
@@ -1,77 +1,90 @@
 package com.stripe.android.paymentsheet.flowcontroller
 
-import com.stripe.android.core.injection.UIContext
 import com.stripe.android.paymentsheet.PaymentSheet
 import com.stripe.android.paymentsheet.analytics.EventReporter
 import com.stripe.android.paymentsheet.state.PaymentSheetLoader
 import com.stripe.android.paymentsheet.state.PaymentSheetState
 import com.stripe.android.paymentsheet.validate
-import kotlinx.coroutines.currentCoroutineContext
-import kotlinx.coroutines.isActive
-import kotlinx.coroutines.withContext
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.CompletableJob
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
 import java.security.InvalidParameterException
 import javax.inject.Inject
 import javax.inject.Singleton
-import kotlin.coroutines.CoroutineContext
 
 @Singleton
 internal class FlowControllerConfigurationHandler @Inject constructor(
     private val paymentSheetLoader: PaymentSheetLoader,
-    @UIContext private val uiContext: CoroutineContext,
     private val eventReporter: EventReporter,
     private val viewModel: FlowControllerViewModel,
     private val paymentSelectionUpdater: PaymentSelectionUpdater,
 ) {
 
-    suspend fun configure(
+    private var job: CompletableJob? = null
+
+    fun configure(
+        scope: CoroutineScope,
         initializationMode: PaymentSheet.InitializationMode,
         configuration: PaymentSheet.Configuration?,
         callback: PaymentSheet.FlowController.ConfigCallback,
+    ) {
+        job?.cancel()
+
+        job = Job().apply {
+            invokeOnCompletion { error ->
+                if (error !is CancellationException) {
+                    callback.onConfigured(
+                        success = error == null,
+                        error = error,
+                    )
+                    job = null
+                }
+            }
+        }
+
+        scope.launch(job!!) {
```

Comment:
Should we model this as deferred instead of job?

`scope.async` instead?

Actually, I'm wondering why we need any of that 🤔 launch returns a job we can later cancel as well.